bpf/flows_sock.c

```diff
@@ -217,6 +212,22 @@ int socket__http_filter(struct __sk_buff *skb) {
             .flags = flags, 
         };
 
+        // Calculate direction based on first flag received
+        // SYN and ACK mean someone else initiated the connection and this is the INGRESS direction
+        if((flags & (SYN_FLAG | ACK_FLAG)) == (SYN_FLAG | ACK_FLAG)) {
+            new_flow.direction = INGRESS;
+        }
+        // SYN only means we initiated the connection and this is the EGRESS direction
+        else if((flags & SYN_FLAG) == SYN_FLAG) {
+            new_flow.direction = EGRESS;
+        // fallback for persistent connections and long-lived connections (f.e. SSH session)
+        } else {
```

Comment:
Unfortunately fallback is necessary here because user space has an evicting flows approach.
For persistent and long-lived connections(like SSH sessions), we don't get packets with SYN or SYNACK flags after eviction. Connection is running, there is no option to guess direction based on TCP flags.